### PR TITLE
Update Ubuntu 23 to 24, update python, make docker images platformed

### PR DESCRIPTION
**Pull request by Hoverbear**
Updates our docker images in a few ways:

* Moves `docker-23` to `docker-24` from Ubuntu 23.04 to Ubuntu 24.04
* Updates Python on `ubuntu-20` to the Python used in `ubuntu-24` (3.12.3)
* Makes our Docker images specify their platform such that aarch64 hosts can use their binfmt/rosetta support to build the respective x86_64 images successfully.
* Moves `ubuntu-24`'s AWS tool to their official build as Ubuntu has moved to a snap.

This should be relevant to @Urhengulas as he's been wanting some of the new Python features like `case` (I have been too!).

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

File: ferrocene/ci/docker-images/ubuntu-20/Dockerfile
Comment (tshepang) [inline ferrocene/ci/docker-images/ubuntu-20/Dockerfile:1]: there are still doc mentions of 22.04

File: ferrocene/ci/docker-images/ubuntu-20/Dockerfile [hunk 1]
```diff
@@ -63,18 +63,18 @@ RUN apt-get update \
 # Grab the newer QEMU from Ubuntu 22.04
 # This requires the binfmt-support package, which is normally a dependency of
 # qemu-user-static.
-COPY --from=jammy /usr/bin/qemu-aarch64-static /usr/bin/qemu-aarch64-static
-COPY --from=jammy /usr/libexec/qemu-binfmt/aarch64-binfmt-P /usr/libexec/qemu-binfmt/aarch64-binfmt-P
-COPY --from=jammy /usr/share/binfmts/qemu-aarch64 /usr/share/binfmts/qemu-aarch64
+COPY --from=noble /usr/bin/qemu-aarch64-static /usr/bin/qemu-aarch64-static
+COPY --from=noble /usr/libexec/qemu-binfmt/aarch64-binfmt-P /usr/libexec/qemu-binfmt/aarch64-binfmt-P
+# COPY --from=noble /usr/share/binfmts/qemu-aarch64 /usr/share/binfmts/qemu-aarch64
```
Comment (pietroalbini) [inline ferrocene/ci/docker-images/ubuntu-20/Dockerfile:68]: Why is this commented?

**Review by pietroalbini — APPROVED** [review]

File: ferrocene/ci/docker-images/ubuntu-20/Dockerfile (hunk 1 above)
Comment (Hoverbear) [reply ferrocene/ci/docker-images/ubuntu-20/Dockerfile:68]: The file doesn't exist anymore, seems I need to hunt for it.

**Comment by Hoverbear** [comment]
bors try

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**